src/en/news/blog/2023/ceph-encryption-performance/index.md

```diff
@@ -0,0 +1,226 @@
+
+
+```
+ms_client_mode = secure
+ms_cluster_mode = secure
+ms_service_mode = secure
+ms_mon_client_mode = secure
+ms_mon_cluster_mode = secure
+ms_mon_service_mode = secure
+```
+
+Three sets of tests were performed as follows with the cluster rebuilt between each set of tests:
+
+Test Set | Client Processes | io_depth
+-- | -- | --
+Multi-client | 10 fio processes per node, 5 client nodes | 128 per process
+Single-client | 1 fio process on 1 client node | 128
+Single-client sync | 1 fio process on 1 client node | 1
+
+OSDs were allowed to use all cores on the nodes.  FIO was configured to first pre-fill RBD volume(s) with large writes, followed by 4MB and 4KB IO tests for 300 seconds each. Certain background processes, such as scrub, deep scrub, PG autoscaling, and PG balancing were disabled.  Finally, an RBD pool with a static 16384 PGs (higher than typically recommended) and 3x replication was used.
+
+# Ceph LUKS Tuning - 4MB IOs
+
+Ceph utilizes a tool called [LUKS](https://en.wikipedia.org/wiki/Linux_Unified_Key_Setup) to encrypt the block device(s) that BlueStore writes data to.  There are several tuning options available that may help improve performance.  Before diving into the full set of tests, let's look at a couple of those options.
+
+Option | Description
+-- | --
+`--perf-submit_from_crypt_cpus` | Disable offloading writes to a separate thread after encryption.  There are some situations where offloading write bios from the encryption threads to a single thread degrades performance significantly.  The default is to offload write bios to the same thread.  This option is only relevant for open action. <br><br>NOTE:  This option is available only for low-level dm-crypt performance tuning, use only if you need a change to default dm-crypt behavior.  Needs kernel 4.0 or later.
+`--sector-size <bytes>` | Set sector size for use with disk encryption.  It must be a power of two and in range 512 - 4096 bytes. The default is 512 bytes sectors.  This option is available only in the LUKS2 mode.
+`--perf-no_read_workqueue`<br>`--perf-no_write_workqueue` | Bypass  dm-crypt internal workqueue and process read or write requests synchronously.  This option is only relevant for open action.<br><br>NOTE:  These options are available only for low-level dm-crypt performance tuning, use only if you need a change to default dm-crypt behavior. Needs kernel 5.9 or later.
+
+Unfortunately, the kernel available in Centos Stream 8 does not support disabling the read and write workqueues. Thus We were not able to test those options.  We were however able to test the others and did so with an initial focus on the multi-client test configuration. 
```

Comment:
wha?  The workqueues are the whole point of this ask.

https://blog.cloudflare.com/speeding-up-linux-disk-encryption/

```The main patch from this blog (in a slightly updated form) has been [merged](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/drivers/md/dm-crypt.c?id=39d42fa96ba1b7d2544db3f8ed5da8fb0d5cb877) into mainline Linux kernel and is available since version 5.9 and onwards. ```

I see Stream 8 is still stuck on 4.18.x

How difficult would it be to 

* Install, say, https://elrepo.org/linux/kernel/el8/x86_64/RPMS/kernel-ml-6.6.1-1.el8.elrepo.x86_64.rpm
* Disable mitigations (or not)
* Run new numbers?